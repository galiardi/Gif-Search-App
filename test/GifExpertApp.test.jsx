import { render, screen, fireEvent, waitFor } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'
describe('Testing src/GifExpertApp', () => { 
 
 test('debe hacer match con el snapshot', () => {
   
  const { container } = render(<GifExpertApp />);
  expect(container).toMatchSnapshot();
  
 });
 
 test('debe renderizar una nueva categoria segun inputValue', async() => {
   const inputValue = 'friends';

   render(<GifExpertApp />);

   const input = screen.getByRole('textbox');
   const form = screen.getByRole('form');
   fireEvent.input(input, {taget: {value: inputValue}});
   fireEvent.submit(form);

  //  await waitFor(() => {
  //   expect(screen.getByText(inputValue));
  //  });

   screen.debug();
 });
 
 });