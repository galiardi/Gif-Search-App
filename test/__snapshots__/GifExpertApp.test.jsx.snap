// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing src/GifExpertApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Busca gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`Testing src/GifExpertApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Busca gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
